src/App.tsx

```diff
@@ -1,24 +1,157 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useState } from "react";
+import { UserWarning } from "./UserWarning";
+import { getTodos, postTodo, deleteTodo } from "./api/todos";
+import { TodoInterface } from "./types/todo";
+import { TodoList } from "./Components/TodoList/TodoList";
+import { Error } from "./Components/Error/Error";
+import { FilterTodo } from "./Components/FilterTodo/FilterTodo";
+import { FilterStatus } from "./types/FilterStatus";
 
-const USER_ID = 0;
+const USER_ID = 6429;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<TodoInterface[]>([]);
+  const [error, setError] = useState("");
+  const [title, setTitle] = useState("");
+  const [temporaryTodo, setTemporaryTodo] = useState<TodoInterface>();
+  const [filter, setFilter] = useState<FilterStatus>(FilterStatus.all);
+  const activeTodosСount = useMemo(() => {
+    return todos.filter(({ completed }) => !completed).length;
+  }, [todos]);
+  const completedTodos = useMemo(() => {
+    return todos.filter(({ completed }) => completed);
+  }, [todos]);
+
+  useEffect(() => {}, [error]);
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((result: React.SetStateAction<TodoInterface[]>) => setTodos(result))
+      .catch(() => setError("Unable to load the todos"));
+  }, []);
+
+  const currentTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filter) {
+        case FilterStatus.active:
+          return !todo.completed;
+
+        case FilterStatus.completed:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+  }, [todos, filter]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const addTodo = (todoTitle: string) => {
+    const newTodo = {
+      title: todoTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTemporaryTodo({ ...newTodo, id: 0 });
+
+    postTodo(USER_ID, newTodo)
+      .then((result) => {
+        setTodos((state) => [...state, result]);
+      })
+      .catch(() => {
+        setError("Unable to add a todo");
+        setTimeout(() => {
+          setError("");
+        }, 3000);
```

Comment:
You should clear all timers on the component unmount
https://bobbyhadz.com/blog/react-cleartimeout